src/EditorFeatures/Core/Shared/Extensions/SmartIndentExtensions.cs

```diff
@@ -6,9 +6,9 @@
 
 namespace Microsoft.CodeAnalysis.Editor.Shared.Extensions
 {
-    internal static class SmartIndentExtensions
+    internal static class IndentationResultExtensions
     {
-        public static int GetIndentation(this IndentationResult result, ITextView textView, ITextSnapshotLine lineToBeIndented)
+        public static int GetIndentation(this Indentation.IndentationResult result, ITextView textView, ITextSnapshotLine lineToBeIndented)
         {
             var position = new SnapshotPoint(lineToBeIndented.Snapshot, result.BasePosition);
             var pointInSurfaceSnapshot = textView.BufferGraph.MapUpToSnapshot(position, PointTrackingMode.Positive, PositionAffinity.Successor, textView.TextSnapshot);
```

Comment:
hmmm.. so ISmartIndenter works on top level buffer rather than subject buffer?

Reply:
why is this desirable behavior ?

 if (!pointInSurfaceSnapshot.HasValue)
            {	            {
                return position.GetContainingLine().GetColumnOfFirstNonWhitespaceCharacterOrEndOfLine(textView.Options);	                return position.GetContainingLine().GetColumnOfFirstNonWhitespaceCharacterOrEndOfLine(textView.Options);
            }	            }

Reply:
I guess basically I am not sure what this GetIndentation is trying to do..